src/os/bluestore/bluefs_types.h

```diff
@@ -59,51 +59,55 @@ struct bluefs_fnode_delta_t {
 
   DENC_HELPERS
 
-  void bound_encode(size_t& p) const {
-    _denc_friend(*this, p);
-  }
-  void encode(ceph::buffer::list::contiguous_appender& p) const {
-    DENC_DUMP_PRE(bluefs_fnode_t);
-    _denc_friend(*this, p);
-  }
-  void decode(ceph::buffer::ptr::const_iterator& p) {
-    DENC_START(2, 2, p);
+  template <typename P>
+  void encode_helper(P& p) const {
+    __u8 need_v = type == WAL_V2 ? 2 : 1;
+    DENC_START(need_v, need_v, p);
     denc_varint(ino, p);
     denc_varint(size, p);
     denc(mtime, p);
     denc(offset, p);
+    if (struct_v >= 2) {
+      denc(type, p);
+    } else {
+      // v1 had a hint field that is no longer used
+      uint8_t __unused__ = 0;
+      denc(__unused__, p);
+    }
```

Comment:
this is basically the same as:
```suggestion
    if (struct_v >= 2) {
      denc(type, p);
    }
```

did you do it that way to signal what the previous implementation was?

Reply:
yes.